packages/solana/src/schemas.ts

```diff
@@ -0,0 +1,116 @@
+import * as base58 from '@chainflip/utils/base58';
+import { assetContractId, chainContractId } from '@chainflip/utils/chainflip';
+import { POLKADOT_SS58_PREFIX } from '@chainflip/utils/consts';
+import * as ss58 from '@chainflip/utils/ss58';
+import anchor from '@coral-xyz/anchor';
+import { z } from 'zod';
+import { tryDecodeCfParams } from './scale';
+
+const { BN } = anchor;
+
+const entries = Object.entries as <T>(o: T) => [keyof T, T[keyof T]][];
+
+const swapParams = z
+  .object({
+    amount: z.instanceof(BN).transform((bn) => bn.toString()),
+    dst_chain: z.number(),
+    dst_address: z.instanceof(Buffer),
+    dst_token: z.number(),
+    ccm_parameters: z
+      .object({
+        message: z.instanceof(Buffer),
+        gas_amount: z.instanceof(BN).transform((bn) => bn.toString()),
+      })
+      .nullable(),
+    cf_parameters: z.instanceof(Buffer).transform((buf) => new Uint8Array(buf)),
+  })
+  .transform((data, ctx) => {
+    const asset = entries(assetContractId).find(([, id]) => id === data.dst_token)?.[0];
+
+    let success = true;
+
+    if (!asset) {
+      ctx.addIssue({
+        message: 'Failed to get destination asset',
+        code: 'custom',
+        path: ['dst_token'],
+      });
+      success = false;
+    }
+
+    const chain = entries(chainContractId).find(([, id]) => id === data.dst_chain)?.[0];
+
+    if (!chain) {
+      ctx.addIssue({
+        message: 'Failed to get destination chain',
+        code: 'custom',
+        path: ['dst_chain'],
+      });
+      success = false;
+    }
+
+    let destinationAddress: string;
+
+    switch (chain) {
+      case 'Arbitrum':
+      case 'Ethereum':
+        destinationAddress = `0x${data.dst_address.toString('hex')}`;
+        break;
+      case 'Bitcoin':
+        destinationAddress = data.dst_address.toString('utf8');
+        break;
+      case 'Polkadot':
+        destinationAddress = ss58.encode({
+          data: data.dst_address,
+          ss58Format: POLKADOT_SS58_PREFIX,
+        });
+        break;
+      case 'Solana':
+        destinationAddress = base58.encode(data.dst_address);
+        break;
+      default: {
+        // already collecting errors
+        const _: undefined = chain;
```

Comment:
dont understand this one

Reply:
At this point, if `chain` is undefined, we are already in an error state. Instead of immediately throwing the first error, we keep parsing and collect all the errors we can find so we can report all errors instead of just the first one.